packages/fetchai/connections/local/connection.py

```diff
@@ -104,13 +104,14 @@ def __init__(
         self._loop = loop if loop is not None else asyncio.new_event_loop()
         self._thread = Thread(target=self._run_loop, daemon=True)
 
-        self._in_queue = asyncio.Queue(loop=self._loop)  # type: asyncio.Queue
+        self._in_queue = None  # type: Optional[asyncio.Queue]
```

Comment:
Are the logic changes necessary?

Reply:
Most of the changes I did where ported from this PR: https://github.com/fetchai/agents-aea/pull/2679/files I didn't gave enough thought, If you want I can take a deeper look why these changes were made